### DISCUSSION TITLE
How can I style the last hline of a table programmatically?

### DISCUSSION THREAD

**Discussion by maxigaz** [PgBiel/typst-tablex#117] [Q&A] [answered]
I would like to define a table template where the first two and the last hlines are black while the rest of the hlines are grey, like in the screenshot:

![typst-tablex-hline-style](https://github.com/PgBiel/typst-tablex/assets/10419911/1fb6fc44-6715-482a-8451-17f3c653afbc)

This is how I've been able to achieve that:

```typ
#import "@preview/tablex:0.0.7": tablex, hlinex, cellx

#let ktable(..cells) = tablex(
  auto-vlines: false,

  map-hlines: v => (
    if v.y > 1 and v.y < 5 {
      return (..v, stroke: color.rgb("#CCC"))
    }
    else {
      return (..v, stroke: color.rgb("#000"))
    }
  ),

  columns: (auto, auto, auto),
  header-rows: 1,

  [*Col 1*],
  [*Col 2*],
  [*Col 3*],

  ..cells,

)

#ktable(
  [#lorem(2)],
  [#lorem(2)],
  [#lorem(2)],

  [#lorem(2)],
  [#lorem(2)],
  [#lorem(2)],

  [#lorem(2)],
  [#lorem(2)],
  [#lorem(2)],

  [#lorem(2)],
  [#lorem(2)],
  [#lorem(2)],
)
```

This piece of code works, but only if there are exactly five rows in the table, or else the condition `v.y < 5` needs to be rewritten for every table instance.

I tried replacing it with a method like `cells.last()` (mentioned in the examples in the README), but there's either no such thing for hlines or I'm not using it correctly.

**Comment by PgBiel** [comment, marked as answer]
Try this:
```typ

#let ktable(..cells) = tablex(
  auto-vlines: false,

  map-hlines: h => {
    if h.stroke == auto {
      h.stroke = color.rgb("#CCC")
    }
    h
  },

  columns: (auto, auto, auto),
  header-rows: 1,

  [*Col 1*],
  [*Col 2*],
  [*Col 3*],

  ..cells,
  hlinex(stroke: color.rgb("#000"))
)
```

**Reply by maxigaz** [reply]
It works. Thank you!
